.github/workflows/publish-providers.yml

```diff
@@ -13,12 +21,171 @@ on:
+            echo "Manual workflow dispatch detected"
+            
+            if [ -n "${{ inputs.packages }}" ]; then
+              # Parse comma-separated input
+              IFS=',' read -ra SPECIFIED_PACKAGES <<< "${{ inputs.packages }}"
+              for pkg in "${SPECIFIED_PACKAGES[@]}"; do
+                # Trim whitespace
+                pkg=$(echo "$pkg" | xargs)
+                # Validate package exists
+                if [[ " ${ALL_PACKAGES[@]} " =~ " ${pkg} " ]]; then
+                  PACKAGES=$(echo "$PACKAGES" | jq -c --arg pkg "$pkg" '. += [$pkg]')
+                  HAS_CHANGES=true
+                  echo "Manual publish requested for: $pkg"
+                else
+                  echo "Warning: Unknown package '$pkg' specified, skipping"
+                fi
+              done
+            else
+              # No packages specified, publish auto-publishable packages only
+              # code-analyzer is excluded from bulk operations and must be specified explicitly
+              for pkg in mcp-provider-api mcp-provider-dx-core; do
+                PACKAGES=$(echo "$PACKAGES" | jq -c --arg pkg "$pkg" '. += [$pkg]')
+                HAS_CHANGES=true
+                echo "Manual publish requested for: $pkg"
+              done
+            fi
+          else
+            echo "Push event detected, checking for changed files"
+            # Get changed files in the last commit
+            CHANGED_FILES=$(git diff --name-only HEAD~1 HEAD)
+            
+            # Check each provider package for changes (only auto-publish enabled packages)
+            for pkg in mcp-provider-api mcp-provider-dx-core; do
+              if echo "$CHANGED_FILES" | grep -q "^packages/$pkg/"; then
+                PACKAGES=$(echo "$PACKAGES" | jq -c --arg pkg "$pkg" '. += [$pkg]')
+                HAS_CHANGES=true
+                echo "Detected changes in: $pkg"
+              fi
+            done
+          fi
+          
+          echo "packages=$PACKAGES" >> "$GITHUB_OUTPUT"
+          echo "has-changes=$HAS_CHANGES" >> "$GITHUB_OUTPUT"
+          echo "Final packages to process: $PACKAGES"
+
+  publish-package:
+    needs: detect-changes
+    if: needs.detect-changes.outputs.has-changes == 'true'
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        package: ${{ fromJson(needs.detect-changes.outputs.changed-packages) }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.SVC_CLI_BOT_GITHUB_TOKEN }}
+          fetch-depth: 0
+
+      - name: Get Github user info
+        id: github-user-info
+        uses: salesforcecli/github-workflows/.github/actions/getGithubUserInfo@main
+        with:
+          SVC_CLI_BOT_GITHUB_TOKEN: ${{ secrets.SVC_CLI_BOT_GITHUB_TOKEN }}
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 'lts/*'
+          cache: yarn
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
```

Comment:
Note for the future: We may want to use https://github.com/salesforcecli/github-workflows/blob/main/.github/actions/yarnInstallWithRetries/action.yml

We could add a `--frozen-lockfile` input